packages/playwright-core/src/server/firefox/ffBrowser.ts

```diff
@@ -331,7 +331,12 @@ export class FFBrowserContext extends BrowserContext {
 
   async doSetHTTPCredentials(httpCredentials?: types.Credentials): Promise<void> {
     this._options.httpCredentials = httpCredentials;
-    await this._browser._connection.send('Browser.setHTTPCredentials', { browserContextId: this._browserContextId, credentials: httpCredentials || null });
+    let credentials = null;
+    if (httpCredentials){
+      credentials = { username: httpCredentials.username, password: httpCredentials.password };
+      if (httpCredentials.origin) (credentials as any).origin = httpCredentials.origin;
```

Comment:
lets get rid of this as any hack? It should be now in the types. Maybe revert the changes in this file and only add the origin in line 334.

Reply:
You're right. I reverted that code. Not sure what you mean by still add the origin in line 334. It seems to work fine without it.